### PR TITLE
Adjust the logic that renders fields with choices on an API Options call

### PR DESCRIPTION
**Pull request by jbradberry**
##### SUMMARY
Previously, non-relational choice fields that had `blank=True` and/or `null=True` set would fail to show this implicit choice when doing an OPTION call to the API endpoint for the resource.  This change adds logic similar to how Django includes this option in when rendering such fields.

related #5099

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
awx: 9.1.0
```

### CODE REVIEW

File: awx/api/metadata.py [hunk 1]
```diff
@@ -96,7 +97,13 @@ def get_field_info(self, field):
             field_info['children'] = self.get_serializer_info(field)
 
         if not isinstance(field, (RelatedField, ManyRelatedField)) and hasattr(field, 'choices'):
-            field_info['choices'] = [(choice_value, choice_name) for choice_value, choice_name in field.choices.items()]
+            choices = [
+                (choice_value, choice_name) for choice_value, choice_name in field.choices.items()
+            ]
+            if not any(choice in ('', None) for choice, _ in choices):
+                if field.allow_blank or (field.allow_null and not isinstance(field, ChoiceNullField)):
+                    choices = [("", "---------")] + choices
```
Comment (AlanCoding) [inline awx/api/metadata.py:105]: Is the empty string accepted in these cases? If so then this looks good to me.

**Comment by wenottingham** [comment]
Is there a reason this PR is in limbo since December?

**Comment by jbradberry** [comment]
@wenottingham yeah, I kind of let it fall off my radar since there were some problems with CI at the time and this is only somewhat related to my main issue.  That said, I think I can get it fixed up without too much difficulty.

**Review by ryanpetrello — APPROVED** [review]
This seems pretty reasonable to me.

Before:

image

After:

image

Also, re: @AlanCoding's original issue (https://github.com/ansible/awx/issues/5099), while I didn't actually *test* w/ send/receive, I expect the issue should be resolved by this PR:

image

**Comment by jbradberry** [comment]
@ryanpetrello thanks for the review and merge!

**Pull request merged**